clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+    assertThat(clusters).hasSize(0);
+  }
+
+  @Test
+  void getSingleCluster(SoftAssertions softly) {
+    KubernetesV2Cluster cluster =
+        clusterProvider.getCluster("frontendapp", ACCOUNT_NAME, "deployment frontend");
+    assertThat(cluster).isNotNull();
+    assertFrontendCluster(softly, cluster, false);
+  }
+
+  @Test
+  void getSingleClusterWithDetails(SoftAssertions softly) {
+    KubernetesV2Cluster cluster =
+        clusterProvider.getCluster("frontendapp", ACCOUNT_NAME, "deployment frontend", true);
+    assertThat(cluster).isNotNull();
+    assertFrontendCluster(softly, cluster, false);
+  }
+
+  @Test
+  void getSingleClusterWrongApp(SoftAssertions softly) {
+    KubernetesV2Cluster cluster =
+        clusterProvider.getCluster("backendapp", ACCOUNT_NAME, "deployment frontend");
+    assertThat(cluster).isNull();
+  }
+
+  @Test
+  void getClusterSummaries(SoftAssertions softly) {
+    Map<String, Set<KubernetesV2Cluster>> results =
+        clusterProvider.getClusterSummaries("backendapp");
+    assertThat(results).hasSize(1);
+    assertThat(results).containsKey(ACCOUNT_NAME);
+
+    Set<KubernetesV2Cluster> clusters = results.get(ACCOUNT_NAME);
+    assertThat(clusters).hasSize(2);
+
+    assertThat(clusters)
+        .extracting(KubernetesV2Cluster::getName)
+        .containsExactlyInAnyOrder("service backendlb", "replicaSet backend");
+
+    Map<String, KubernetesV2Cluster> clusterLookup =
+        clusters.stream().collect(toImmutableMap(KubernetesV2Cluster::getName, c -> c));
+
+    assertBackendServiceCluster(softly, clusterLookup.get("service backendlb"));
+    assertBackendCluster(softly, clusterLookup.get("replicaSet backend"), true);
+  }
+
+  @Test
+  void getServerGroup(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(ACCOUNT_NAME, "backend-ns", "replicaSet backend-v014");
+    assertThat(serverGroup).isNotNull();
+    assertBackendPriorServerGroup(softly, serverGroup);
+  }
+
+  @Test
+  void getServerGroupWrongNamespace(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(ACCOUNT_NAME, "frontend-ns", "replicaSet backend-v014");
+    assertThat(serverGroup).isNull();
+  }
+
+  @Test
+  void getServerGroupWithDetails(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(ACCOUNT_NAME, "backend-ns", "replicaSet backend-v014", true);
+    assertThat(serverGroup).isNotNull();
+    assertBackendPriorServerGroup(softly, serverGroup);
+  }
+
+  @Test
+  void getServerGroupWithoutDetails(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(
+            ACCOUNT_NAME, "backend-ns", "replicaSet backend-v014", false);
+    assertThat(serverGroup).isNotNull();
+    // Looks like we ignore the includeDetails flag, so this is the same serverGroup as when we do
```

Comment:
Oh, I think this answers my previous question -- same deal for clusters?

Reply:
Added a comment above, but in the above case I was missing a test; for clusters the flag is respected.